### PR TITLE
Summarize the process of transforming ideas to commands

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/wp-cli/wp-cli/issues/4236

### CODE REVIEW

**Comment by danielbachhuber** [comment]
To help guide feedback, here are some questions:

* After reading this blurb, what questions come to mind?
* What can we add or improve upon in [WP-CLI's philosophy](https://make.wordpress.org/cli/handbook/philosophy/) to clarify the decision-making process?
* Using https://github.com/wp-cli/ideas/issues/45#issuecomment-314254600 as an example, how can we improve the process when someone in the community takes it upon themselves to write a new command that makes sense to adopt into WP-CLI eventually?

**Comment by gitlost** [comment]
> After reading this blurb, what questions come to mind?

None to mine.

> What can we add or improve upon in WP-CLI's philosophy to clarify the decision-making process?

A link back to here now!

> Using #45 (comment) as an example, how can we improve the process...

The process is working well in that case, isn't it?

File: README.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
-Have an idea for a WP-CLI command? [Create a new issue](https://github.com/wp-cli/ideas/issues/new) or [up-vote an existing issue](https://github.com/wp-cli/ideas/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
+### Have an idea for a WP-CLI command?**
+
+Feel free to [up-vote an existing issue](https://github.com/wp-cli/ideas/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) if you can find one that matches, or [create a new issue](https://github.com/wp-cli/ideas/issues/new) if no one has proposed the idea yet.
+
+### Curious as to how ideas become WP-CLI commands?
+
+The process generally works like this:
+
+0. **New issue is created.** Good issues include substantial detail into what the command might do, how it might work, and what value it would add to the ecosystem.
```
Comment (gitlost) [inline README.md]: Maybe "A new issue",

File: README.md [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+1. **Some discussion takes place.** Issues are evaluated against [WP-CLI's philosophy](https://make.wordpress.org/cli/handbook/philosophy/), technical feasibility, and estimated impact. Everyone is free to weigh in; we value thoughtful consideration that introduces new viewpoints.
```
Comment (gitlost) [inline README.md]: Maybe add something like "It's here that the decision is made to either add a new command or add a new subcommand to an existing command, or add the functionality as a new option to an existing command or subcommand."

File: README.md [hunk 3]
```diff
@@ -1,4 +1,16 @@
...
+2. **New WP-CLI commands are created.** In some cases they're written as standalone packages, and in others they're submitted as pull requests to existing repositories. In some cases they're produced by committers, and in others they're developed by members of the community. It's a pretty organic process.
```
Comment (gitlost) [inline README.md]: Maybe
2. Some development takes place. In some cases...
3. Iterate between 1. and 2. It's a pretty organic process.
4. A new WP-CLI command is born/created.

**Comment by danielbachhuber** [comment]
> The process is working well in that case, isn't it?

Well, we've kinda dropped @swissspidy's contribution (e.g. it seems to be languishing at the moment). To get something like `wp oembed *` landed, we'd need to:

* Have documentation we can refer to for coding/implementation standards.
* Ensure there's sufficient test coverage for the implemented features.
* Establish specific criteria for when the package is considered "complete" (or at whatever stage we're going to take some action on it).
* Jump in to help out as needed.

For larger commands, it might make sense to assign a designated buddy, so there's clear responsibility delegated to one person.

**Comment by gitlost** [comment]
Yes, all good, though of course there's always going to be a lot of fall off and stalled and abandoned stuff.

A designed buddy might be good for all contributions that are taking some time.

**Comment by danielbachhuber** [comment]
> Should we have a list page like features plugin tracking?

Good idea. I don't know if we need one right now, but the [roadmap](https://make.wordpress.org/cli/handbook/roadmap/) might eventually be a good place for this.

> It would be nice to guide participation in the meeting. People who have an idea probably want to ask something to us.

Makes sense. What steps do you think we should take at this point?

**Comment by miya0001** [comment]
> Good idea. I don't know if we need one right now, but the roadmap might eventually be a good place for this.

Yeah, I thinks so too.

> What steps do you think we should take at this point?

For example, add a line like following into this documentation.
"We are having a slack meeting at every Tuesday, if you have an idea please tell us about it. :)"
I am not sure this sentence is better or not. 😄

File: README.md [hunk 4]
```diff
@@ -1,4 +1,16 @@
 wp-cli/ideas
 ============
 
-Have an idea for a WP-CLI command? [Create a new issue](https://github.com/wp-cli/ideas/issues/new) or [up-vote an existing issue](https://github.com/wp-cli/ideas/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
+### Have an idea for a WP-CLI command?**
+
+Feel free to [up-vote an existing issue](https://github.com/wp-cli/ideas/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) if you can find one that matches, or [create a new issue](https://github.com/wp-cli/ideas/issues/new) if no one has proposed the idea yet.
```
Comment (schlessera) [inline README.md]: What does "up-voting" actually mean? If we provide a strict procedure, like add a 👍  for up-voting and a 👎 for down-voting, on the initial post of an issue (the first one), we can use this "pseudo-structured" information to build a dashboard with the best ideas.
This dashboard can then serve as the main entry-point for our feature implementation process, showing an overview of our "ideas-roadmap", and providing links to work on the existing one or create a new one.

Comment (danielbachhuber) [inline README.md]: > What does "up-voting" actually mean?

I've changed the language to "use an emoji reaction".

Up-voting probably too strongly implies new feature creation is a purely democratic process (e.g. the most up-votes gets picked). Emoji reactions are better used to gauge level of interest.

File: README.md (hunk 1 above)
Comment (danielbachhuber) [inline README.md]: Good catch, updated a0d99927ff5c157be303390034ddfe1c4109440c

**Pull request merged**